moqt-server/src/modules/handlers/subscribe_error_handler.rs

```diff
@@ -0,0 +1,376 @@
+use anyhow::{bail, Result};
+use moqt_core::pubsub_relation_manager_repository::PubSubRelationManagerRepository;
+use moqt_core::{
+    constants::StreamDirection,
+    messages::{control_messages::subscribe_error::SubscribeError, moqt_payload::MOQTPayload},
+    MOQTClient, SendStreamDispatcherRepository,
+};
+
+pub(crate) async fn subscribe_error_handler(
+    subscribe_error_message: SubscribeError,
+    pubsub_relation_manager_repository: &mut dyn PubSubRelationManagerRepository,
+    send_stream_dispatcher_repository: &mut dyn SendStreamDispatcherRepository,
+    client: &mut MOQTClient,
+) -> Result<()> {
+    tracing::trace!("subscribe_error_handler start.");
+
+    tracing::debug!("subscribe_error_message: {:#?}", subscribe_error_message);
+
+    let upstream_session_id = client.id;
+    let upstream_subscribe_id = subscribe_error_message.subscribe_id();
+
+    // TODO: Retry to send the SUBSCRIBE message if the error type is Retry Track Alias
+    // Determine the SUBSCRIBER who sent the SUBSCRIBE using the track_namespace and track_name
+    let downstream_ids = pubsub_relation_manager_repository
+        .get_requesting_downstream_session_ids_and_subscribe_ids(
+            upstream_subscribe_id,
+            upstream_session_id,
+        )
+        .await?;
+    match downstream_ids {
+        Some(downstream_ids) => {
+            // Notify all waiting subscribers with the subscribe_error message
+            // And delete subscripsions and relation
+            for (downstream_session_id, downstream_subscribe_id) in downstream_ids.iter() {
+                match delete_downstream_and_upstream_subscription(
+                    pubsub_relation_manager_repository,
+                    upstream_session_id,
+                    upstream_subscribe_id,
+                    *downstream_session_id,
+                    *downstream_subscribe_id,
+                )
+                .await
+                {
+                    Ok(_) => {}
+                    Err(e) => {
+                        bail!("cannot delete publisher and subscriber: {:?}", e);
+                    }
+                }
+
+                let mut relaying_subscribe_error_message = subscribe_error_message.clone();
```

Comment:
It is s very small point.

Two SUBSCRIBE_ERROR messaage are similar but different in subscribe_id.
- Producer(Client) -> Consumer(Server)
- Producer(Server) -> Consumer(Client)

So, I think it is better to use `new()` than to use `clone()`.

Reply:
OK.

I think subscribe and subscribe_ok has same problem too.
Can we close this PR and fix them in the other PR?

Reply:
ok